test/jdk/javax/swing/JButton/4796987/bug4796987.java

```diff
@@ -26,7 +26,8 @@
  * @bug 4796987
  * @key headful
  * @requires (os.family == "windows")
- * @summary XP Only: JButton.setBorderPainted() does not work with XP L&F
+ * @summary Verify JButton.setBorderPainted(false) removes border
+ *      for Windows visual styles (Windows XP and later)
```

Comment:
Is it necessary to mention about Windows XP ?

Reply:
It just references the point where the feature was introduced.

Currently supported versions of Java likely don't start on Windows XP any more, but it doesn't change the fact that visual styles were introduced in Windows XP.

Reply:
Ok.